packages/shuttle/src/shuttle/onChainEventReconciliation.ts

```diff
@@ -0,0 +1,249 @@
+import {
+  ClientDuplexStream,
+  HubRpcClient,
+  OnChainEvent,
+  OnChainEventType,
+  IdRegisterEventBody,
+  SignerEventBody,
+  StorageRentEventBody,
+} from "@farcaster/hub-nodejs";
+import { type DB } from "./db";
+import { pino } from "pino";
+import { ok } from "neverthrow";
+import { extendStackTrace } from "../utils";
+
+const MAX_PAGE_SIZE = 500;
+
+export type DBOnChainEvent = {
+  id: string;
+  chainId: bigint;
+  createdAt: Date;
+  updatedAt: Date;
+  blockTimestamp: Date;
+  fid: number;
+  blockNumber: bigint;
+  logIndex: number;
+  type: OnChainEventType;
+  txHash: Uint8Array;
+  body: IdRegisterEventBody | SignerEventBody | StorageRentEventBody;
```

Comment:
```suggestion
  body: IdRegisterEventBody | SignerEventBody | SignerMigratedEventBody | StorageRentEventBody;
```

For completeness, since by default backfill does request this type of on-chain event from the hub (https://github.com/neynarxyz/hub-monorepo/pull/2/files#diff-8b92dfe3afde86f73f1479c2a6a5ad501444213839bf49202a0fdd161d8e6408R59)